FA.py

```diff
@@ -0,0 +1,97 @@
+from Transition import *
+
+
+class FA:
+    def __init__(self):
+        self.initial_state = ""
+        self.states = []
+        self.alphabet = []
+        self.final_states = []
+        self.transitions = []
+
+    def reader(self, filename):
+        file = open(filename, 'r')
+        line = file.readline()
+
+        # Denote with 'step' the current line we are in
+        # if state is 1, it means we are reading the states, and so on
+        step = 0
+
+        while line != "":
+            step += 1
+
+            if step == 1:
+                # read the sates
+                line = line.split()
+                self.states = line
+                self.initial_state = self.states[0]
```

Comment:
So the first state represents the initial state? You should mention this or read the initial state from another line